processor/k8sattributesprocessor/e2e_test.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2020 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package k8sattributesprocessor
+
+import (
+	"bufio"
+	"errors"
+	"io"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.opentelemetry.io/collector/pdata/ptrace"
+)
+
+func readJSONMessage(br *bufio.Reader) ([]byte, bool, error) {
+	buf, _, c := br.ReadLine()
+	if c == io.EOF {
+		return nil, true, nil
+	}
+	return buf, false, nil
+}
+
+func TestJobE2E(t *testing.T) {
+	if _, err := os.Stat("./testdata/trace.json"); errors.Is(err, os.ErrNotExist) {
+		return
+	}
+
+	f, err := os.Open("./testdata/trace.json")
+	assert.NoError(t, err)
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+
+	const maxCapacity int = 8388608
+	buf := make([]byte, maxCapacity)
+	scanner.Buffer(buf, maxCapacity)
+
+	unmarshaler := ptrace.JSONUnmarshaler{}
+	jobKeyFound := false
+	nsKeyFound := false
+	for scanner.Scan() {
+		traces, err := unmarshaler.UnmarshalTraces(scanner.Bytes())
+		assert.NoError(t, err)
+
+		len := traces.ResourceSpans().Len()
+		for i := 0; i < len; i++ {
+			resourceSpans := traces.ResourceSpans().At(i)
+
+			t.Logf("resource attributes: %s", resourceSpans.Resource().Attributes().AsRaw())
+
+			resourceSpans.Resource().Attributes().Range(
+				func(k string, v pcommon.Value) bool {
+					if k == "k8s.job.name" && v.Str() == "test-telemetrygen" {
+						jobKeyFound = true
```

Comment:
Should we count here to make sure that every span has the keys? `jobAttrCount++`